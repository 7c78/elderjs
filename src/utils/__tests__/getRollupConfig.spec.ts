/* eslint-disable global-require */
import multiInput from 'rollup-plugin-multi-input';
import path from 'path';
import { createBrowserConfig, createSSRConfig } from '../getRollupConfig';

describe('#getRollupConfig', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('createBrowserConfig works', () => {
    [true, false].forEach((sourcemap) => {
      const { plugins, ...config } = createBrowserConfig({
        input: [`./components/*/*.svelte`],
        output: {
          dir: './public/dist/svelte/',
          entryFileNames: 'entry[name]-[hash].js',
          sourcemap,
          format: 'system',
        },
        multiInputConfig: multiInput({
          // TODO: test with false
          relative: `./components`,
          transformOutputPath: (output) => `${path.basename(output)}`,
        }),
        svelteConfig: {},
      });
      expect(config).toEqual({
        cache: true,
        input: ['./components/*/*.svelte'],
        output: {
          dir: './public/dist/svelte/',
          entryFileNames: 'entry[name]-[hash].js',
          format: 'system',
          sourcemap,
        },
        treeshake: true,
      });
      expect(plugins).toHaveLength(8);
    });
  });

  it('createBrowserConfig multiInputConfig = false', () => {
    expect(
      createBrowserConfig({
        input: [`./components/*/*.svelte`],
        output: {
          dir: './public/dist/svelte/',
          entryFileNames: 'entry[name]-[hash].js',
          sourcemap: true,
          format: 'system',
        },
        svelteConfig: {},
        multiInputConfig: false,
      }).plugins.map((p) => p.name),
    ).toEqual(['replace', 'json', 'svelte', 'node-resolve', 'commonjs', 'babel', 'terser']);
  });

  it('createSSRConfig works', () => {
    const { plugins, ...config } = createSSRConfig({
      input: [`./components/*/*.svelte`],
      output: {
        dir: './___ELDER___/compiled/',
        format: 'cjs',
        exports: 'auto',
      },
      multiInputConfig: multiInput({
        relative: `./components`,
        transformOutputPath: (output) => `${path.basename(output)}`,
      }),
      svelteConfig: {
        preprocess: [
          {
            style: ({ content }) => {
              return content.toUpperCase();
            },
          },
        ],
      },
    });
    expect(config).toEqual({
      cache: true,
      input: ['./components/*/*.svelte'],
      output: {
        dir: './___ELDER___/compiled/',
        exports: 'auto',
        format: 'cjs',
      },
      treeshake: true,
    });

    expect(plugins).toHaveLength(8);
  });

  it('createSSRConfig multiInputConfig = false', () => {
    expect(
      createSSRConfig({
        input: [`./components/*/*.svelte`],
        output: {
          dir: './___ELDER___/compiled/',
          format: 'cjs',
          exports: 'auto',
        },
        svelteConfig: {
          preprocess: [
            {
              style: ({ content }) => {
                return content.toUpperCase();
              },
            },
          ],
        },
        multiInputConfig: false,
      }).plugins.map((p) => p.name),
    ).toEqual(['replace', 'json', 'svelte', 'node-resolve', 'commonjs', 'css', 'terser']);
  });

  it('getPluginPaths works', () => {
    jest.mock('glob', () => ({
      sync: jest
        .fn()
        .mockImplementationOnce(() => ['a-file1.svelte', 'a-file2.svelte'])
        .mockImplementationOnce(() => ['b-file1.svelte', 'b-file2.svelte']),
    }));

    jest.mock('path', () => ({
      resolve: (...strings) => strings.join('/').replace('./', '').replace('//', '/').replace('/./', '/'),
      posix: () => ({ dirname: () => '' }),
    }));

    jest.mock('fs-extra', () => ({
      existsSync: jest
        .fn()
        .mockImplementationOnce(() => true) // first plugin from src
        .mockImplementationOnce(() => false) // 2nd from node modules
        .mockImplementationOnce(() => true),
    }));

    expect(
      // @ts-ignore
      require('../getRollupConfig').getPluginPaths({
        srcDir: './src',
        rootDir: './',
        plugins: {
          pluginA: {},
          pluginB: {},
        },
      }),
    ).toEqual(['src/plugins/pluginA/', '/node_modules/pluginB/']);
  });

  it('getRollupConfig as a whole works', () => {
    jest.mock('../getConfig', () => () => ({
      $$internal: {
        clientComponents: 'test/public/svelte',
        ssrComponents: 'test/___ELDER___/compiled',
      },
      distDir: './dist',
      srcDir: './src',
      rootDir: './',
      plugins: {
        pluginA: {},
        pluginB: {},
      },
    }));

    jest.mock('path', () => ({
      resolve: (...strings) => strings.join('/').replace('./', '').replace('//', '/').replace('/./', '/'),
      posix: () => ({ dirname: () => '' }),
    }));
    jest.mock('del');
    jest.mock('fs-extra', () => ({
      existsSync: jest.fn().mockImplementation(() => true),
    }));
    jest.mock('glob', () => ({
      sync: jest
        .fn()
        .mockImplementationOnce(() => ['route1.svelte', 'route2.svelte'])
        .mockImplementationOnce(() => ['layout1.svelte', 'layout2.svelte'])
        .mockImplementationOnce(() => ['pluginA.svelte', 'pluginB.svelte'])
        .mockImplementationOnce(() => ['foo.svelte', 'bar.svelte']),
    }));

    const svelteConfig = {
      preprocess: [
        {
          style: ({ content }) => {
            return content.toUpperCase();
          },
        },
      ],
    };
    expect(require('../getRollupConfig').default({ svelteConfig })).toMatchSnapshot();
  });
});
