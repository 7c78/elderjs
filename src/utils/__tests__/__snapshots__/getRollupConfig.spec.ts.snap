// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#getRollupConfig getRollupConfig as a whole works - default options 1`] = `
Array [
  Object {
    "input": "./node_modules/intersection-observer/intersection-observer.js",
    "output": Array [
      Object {
        "file": "dist/static/intersection-observer.js",
        "format": "iife",
        "name": "./static/intersection-observer.js",
        "plugins": Array [
          Object {
            "name": "terser",
            "renderChunk": [Function],
          },
        ],
      },
    ],
  },
  Object {
    "cache": true,
    "input": Array [
      "rc/layouts/*.svelte",
      "rc/routes/*/*.svelte",
    ],
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      Object {
        "name": "terser",
        "renderChunk": [Function],
      },
    ],
    "treeshake": true,
  },
  Object {
    "cache": true,
    "input": Array [
      "rc/components/*/*.svelte",
      "rc/components/*.svelte",
    ],
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].mjs",
        "format": "esm",
        "sourcemap": false,
      },
    ],
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "load": [Function],
        "name": "babel",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "name": "terser",
        "renderChunk": [Function],
      },
    ],
    "treeshake": true,
  },
  Object {
    "cache": true,
    "input": Array [
      "rc/components/*/*.svelte",
      "rc/components/*.svelte",
    ],
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      Object {
        "name": "terser",
        "renderChunk": [Function],
      },
    ],
    "treeshake": true,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginA/*.svelte",
    ],
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].js",
        "format": "esm",
        "sourcemap": false,
      },
    ],
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "load": [Function],
        "name": "babel",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "name": "terser",
        "renderChunk": [Function],
      },
    ],
    "treeshake": true,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginA/*.svelte",
    ],
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      Object {
        "name": "terser",
        "renderChunk": [Function],
      },
    ],
    "treeshake": true,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginB/*.svelte",
    ],
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].js",
        "format": "esm",
        "sourcemap": false,
      },
    ],
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "load": [Function],
        "name": "babel",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "name": "terser",
        "renderChunk": [Function],
      },
    ],
    "treeshake": true,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginB/*.svelte",
    ],
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      Object {
        "name": "terser",
        "renderChunk": [Function],
      },
    ],
    "treeshake": true,
  },
]
`;

exports[`#getRollupConfig getRollupConfig as a whole works - splitComponents: true, legacy: true 1`] = `
Array [
  Object {
    "input": "./node_modules/intersection-observer/intersection-observer.js",
    "output": Array [
      Object {
        "file": "dist/static/intersection-observer.js",
        "format": "iife",
        "name": "./static/intersection-observer.js",
        "plugins": Array [
          Object {
            "name": "terser",
            "renderChunk": [Function],
          },
        ],
      },
    ],
  },
  Object {
    "cache": true,
    "input": Array [
      "rc/layouts/*.svelte",
      "rc/routes/*/*.svelte",
    ],
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      false,
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "NestedSrcComponent.svelte",
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].js",
        "format": "esm",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "NestedSrcComponent.svelte",
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      false,
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "SrcComponent.svelte",
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].js",
        "format": "esm",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "SrcComponent.svelte",
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      false,
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "SrcComponent2.svelte",
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].js",
        "format": "esm",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "SrcComponent2.svelte",
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      false,
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginA/*.svelte",
    ],
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].js",
        "format": "esm",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginA/*.svelte",
    ],
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      false,
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "LegacyPluginAComponent3.svelte",
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "iife[name]-[hash].js",
        "format": "iife",
        "name": "___elderjs_undefined",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "load": [Function],
        "name": "babel",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "LegacyPluginAComponent4.svelte",
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "iife[name]-[hash].js",
        "format": "iife",
        "name": "___elderjs_undefined",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "load": [Function],
        "name": "babel",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginB/*.svelte",
    ],
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "entry[name]-[hash].js",
        "format": "esm",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": Array [
      "src/plugins/pluginB/*.svelte",
    ],
    "output": Object {
      "dir": "test/___ELDER___/compiled",
      "exports": "auto",
      "format": "cjs",
    },
    "plugins": Array [
      Object {
        "options": [Function],
        "pluginName": "rollup-plugin-multi-input",
      },
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "name": "css",
        "transform": [Function],
      },
      false,
    ],
    "treeshake": false,
  },
  Object {
    "cache": true,
    "input": "LegacyPluginBComponent3.svelte",
    "output": Array [
      Object {
        "dir": "test/public/svelte",
        "entryFileNames": "iife[name]-[hash].js",
        "format": "iife",
        "name": "___elderjs_undefined",
        "sourcemap": true,
      },
    ],
    "plugins": Array [
      Object {
        "name": "replace",
        "renderChunk": [Function],
        "transform": [Function],
      },
      Object {
        "name": "json",
        "transform": [Function],
      },
      Object {
        "generateBundle": [Function],
        "load": [Function],
        "name": "svelte",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "buildStart": [Function],
        "generateBundle": [Function],
        "getPackageInfoForId": [Function],
        "load": [Function],
        "name": "node-resolve",
        "resolveId": [Function],
      },
      Object {
        "buildStart": [Function],
        "load": [Function],
        "name": "commonjs",
        "resolveId": [Function],
        "transform": [Function],
      },
      Object {
        "load": [Function],
        "name": "babel",
        "resolveId": [Function],
        "transform": [Function],
      },
    ],
    "treeshake": false,
  },
]
`;
